### ISSUE TITLE
Regarding the test results after running the code.

### ISSUE THREAD

**Issue by honeymoon52031** [hou27/ReCDAP-FKGC#4]
Thank you for your valuable work. I have some questions that I would like to ask you for answers to. After completing all the preliminary work, I ran the command for the NELL dataset.The parameters are here：
python main.py --dataset NELL-One \
               --data_path ./NELL \
               --few 5 \
               --data_form Pre-Train \
               --prefix nell_recdap_iter100_lr1e-3_margin1 \
               --device 0 \
               --batch_size 64 \
               --g_batch 1024 \
               --epoch 100000 \
               --eval_epoch 1000 \
               --checkpoint_epoch 1000 \
               --learning_rate 1e-3 \
               --margin 1.0 \
               --num_diffusion_iters 100
However, the terminal results told me that the best training epoch was saved at 16000, and the terminal occurred at 19000. Therefore, I obtained the final metric, which was far lower than the results in your paper.There are the result：

Epoch  19000 has finished, saving...
Epoch  19000 has finished, validating...
979     MRR: 0.210      Hits@10: 0.379  Hits@5: 0.154   Hits@1: 0.120
        Best MRR of valid set is 0.210 at 16000 | bad count is 2
        Early stopping at epoch 19000
Training has finished
        Best epoch is 16000 | MRR of valid set is 0.210
Finish
test
nell_recdap_iter100_lr1e-3_margin1_2022_cuda:0_20251103183446
reload state_dict from state/nell_recdap_iter100_lr1e-3_margin1_2022_cuda:0_20251103183446/state_dict
2103    MRR: 0.280      Hits@10: 0.696  Hits@5: 0.311   Hits@1: 0.120

I'd like to know if this discrepancy is normal. If I want to reproduce your project, can I change the parameter value in my running code to `--early_stopping_patience 10`? Will this affect the final result?Or could you please explain how to run your code to obtain the correct numbers mentioned in your paper?
Thank you very much！！！

**Comment by hou27** [comment]
Thank you for reporting this issue.

Looking at your results (MRR: 0.280), this is considerably lower than anything I've observed in my experiments. As I mentioned in the previous issue (#2), some variance is expected due to the inherited codebase, but I've never encountered results this low with the exact same settings.

Here's my experiment history with same parameters:

MRR | Hits@10 | Hits@5 | Hits@1
-- | -- | -- | --
0.505 | 0.528 | 0.506 | 0.493
0.477 | 0.483 | 0.477 | 0.469
0.467 | 0.492 | 0.476 | 0.451
0.521 | 0.545 | 0.529 | 0.505
0.473 | 0.477 | 0.471 | 0.465
0.502 | 0.520 | 0.505 | 0.490
0.489 | 0.492 | 0.485 | 0.478
0.427 | 0.437 | 0.430 | 0.414

My lowest MRR is 0.427, which is still higher than your result and everything needed to reproduce the paper is in this repository. I hope this helps you.

**Comment by honeymoon52031** [comment]
Okay, thank you so much for your answer.Yesterday I adjusted the command, adding `--early_stopping_patience 10` to increase the patience from 3 to 10, but it seems to have produced even worse results.Below are the adjusted commands and test results.
python main.py --dataset NELL-One \
>                --data_path ./NELL \
>                --few 5 \
>                --data_form Pre-Train \
>                --prefix nell_recdap_iter100_lr1e-3_margin1 \
>                --device 0 \
>                --batch_size 64 \
>                --g_batch 1024 \
>                --epoch 100000 \
>                --eval_epoch 1000 \
>                --checkpoint_epoch 1000 \
>                --learning_rate 1e-3 \
>                --early_stopping_patience 10 \
>                --margin 1.0 \
>                --num_diffusion_iters 100

results：
 MRR: 0.131      Hits@10: 0.265  Hits@5: 0.131   Hits@1: 0.092

Epoch  34000 has finished, saving...
Epoch  34000 has finished, validating...
979     MRR: 0.101      Hits@10: 0.143  Hits@5: 0.106   Hits@1: 0.079
        Best MRR of valid set is 0.140 at 24000 | bad count is 9
        Early stopping at epoch 34000
Training has finished
        Best epoch is 24000 | MRR of valid set is 0.140
Finish

The optimal model was reached at 24,000 epochs, and the process terminated at 34,000 epochs.I'd like to know if your command will run the entire epoch without terminating. If it does terminate, could you explain why? Thank you for your answer！

**Comment by hou27** [comment]
On my side, it usually stops around epoch 6,000~10,000 as I remember. And yes, it does stop before running entire epochs due to the patience setting as you mentioned.

**Comment by honeymoon52031** [comment]
Ok，thank you so much for your answer！Besides that, have you set any other random seeds? Or are you using the default 2022? If so, could you please specify the exact seed? I'm using a RTX4090, and I'd like to know if even slight differences in configuration can have an impact.
